agent/engine/task_manager.go

```diff
@@ -451,6 +473,50 @@ func (mtask *managedTask) startContainerTransitions(transitionFunc containerTran
 	return anyCanTransition, transitions
 }
 
+type containerTransitionFunc func(container *api.Container, nextStatus api.ContainerStatus)
+
+// containerNextState determines the next state a container should go to.
+// It returns: The state it should transition to, a bool indicating whether any
+// action is required, and a bool indicating whether a known status change is
+// possible.
+//
+// 'Stopped, false, true' -> "You can move it to known stopped, but you don't have to call a transition function"
```

Comment:
Are these all the cases? It isn't clear if these are examples or an exhaustive list. In either case, super helpful to have!